3-UsingAPIs/Week2/assignment/ex2-pokemonApp/index.js

```diff
@@ -22,18 +23,86 @@ Use async/await and try/catch to handle promises.
 Try and avoid using global variables. As much as possible, try and use function 
 parameters and return values to pass data back and forth.
 ------------------------------------------------------------------------------*/
-function fetchData(/* TODO parameter(s) go here */) {
-  // TODO complete this function
+async function fetchData(url) {
+  try {
+    const response = await fetch(url);
+    if (!response.ok) {
+      throw new Error('HTTP or network errors');
+    }
+    const pokemon = await response.json();
+    return pokemon;
+  } catch (error) {
+    console.log('HTTP or network errors');
+  }
 }
 
-function fetchAndPopulatePokemons(/* TODO parameter(s) go here */) {
-  // TODO complete this function
+// eslint-disable-next-line no-unused-vars
+async function fetchAndPopulatePokemons() {
+  const contentDiv = document.querySelector('.content');
+  const select = document.querySelector('.select-pokemon');
+  console.log('select', select);
+  try {
+    const pokemons = await fetchData(
+      'https://pokeapi.co/api/v2/pokemon?limit=151'
+    );
+
+    pokemons.results.forEach((pokemon) => {
+      const option = document.createElement('option');
+      option.textContent = pokemon.name;
+      option.value = pokemon.url;
+      select.appendChild(option);
+    });
+
+    contentDiv.appendChild(select);
+  } catch (error) {
+    console.error('Error fetching and populating pokemons:', error);
+  }
 }
 
-function fetchImage(/* TODO parameter(s) go here */) {
-  // TODO complete this function
+// eslint-disable-next-line no-unused-vars
+async function fetchImage() {
+  const selectedPokemonUrl = document.querySelector('.select-pokemon').value;
+
+  if (!selectedPokemonUrl) {
+    throw new Error('Image url in not defined');
+  }
+
+  try {
+    const pokemonResponse = await fetchData(selectedPokemonUrl);
+    const pokemonData = pokemonResponse.sprites.front_default;
+
+    const currentImage = document.getElementById('pokemon-image');
+
+    if (currentImage) {
+      currentImage.src = pokemonData;
+    } else {
+      const contentDiv = document.querySelector('.content');
+      const image = document.createElement('img');
+      image.src = pokemonData;
+      image.setAttribute('id', 'pokemon-image');
+      contentDiv.appendChild(image);
+    }
+  } catch (error) {
+    console.error('Error fetching and displaying image:', error);
+  }
 }
 
-function main() {
-  // TODO complete this function
+async function main() {
+  const contentDiv = document.createElement('div');
+  contentDiv.setAttribute('class', 'content');
+
+  const button = document.createElement('button');
+  button.textContent = 'GET POKEMON!';
+  button.type = 'button';
+  button.setAttribute('onclick', 'fetchAndPopulatePokemons()');
```

Comment:
Don't use attributes such as `onclick` and `onchange` in JavaScript. They only allow a single event handler to be added. Use `addEventListener()` instead:

```suggestion
  button.addEventListener('click', fetchAndPopulatePokemons);
```

Also note that pressing the button multiple times will make the select list longer and longer. Can you think of a fix for this?

Reply:
You did not address this issue:

> Also note that pressing the button multiple times will make the select list longer and longer. Can you think of a fix for this?